### PR TITLE
Added recipe for cabal install

### PR DESCRIPTION
**Pull request by eggzilla**

### CODE REVIEW

**Comment by eggzilla** [comment]
Plz do not merge yet, I need to check on the licence and I want to add a fix to the ghc feedstock.

**Comment by eggzilla** [comment]
I would recommend merging now.

**Comment by eggzilla** [comment]
Ping @bgruening

File: recipes/cabal/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+export CFLAGS="-I$PREFIX/include:$CFLAGS" 
+export LDFLAGS="-L$PREFIX/lib:$LDFLAGS"
+export LD_LIBRARY_PATH="$PREFIX/lib:$LD_LIBRARY_PATH"
+export LIBRARY_PATH="$PREFIX/lib:$LIBRARY_PATH"
+export C_INCLUDE_PATH="$PREFIX/include:$C_INCLUDE_PATH" 
+ghc-pkg recache
+cd cabal-install
+export EXTRA_CONFIGURE_OPTS="--extra-include-dirs=$PREFIX/include --extra-lib-dirs=$PREFIX/lib $EXTRA_CONFIGURE_OPTS";
+echo $EXTRA_CONFIGURE_OPTS
```
Comment (bgruening) [inline recipes/cabal/build.sh]: This can be removed?

File: recipes/cabal/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+build:
+  number: 0
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - ghc
+    - zlib
```
Comment (bgruening) [inline recipes/cabal/meta.yaml]: zlib can be removed from the build section and ghc and gmp probably as well

Comment (eggzilla) [reply recipes/cabal/meta.yaml]: GHC is needed for building, zlib is needed for linking in zlib.h, as well as libgmp.10.so

File: recipes/cabal/build.sh (hunk 1 above)
Comment (eggzilla) [reply recipes/cabal/build.sh]: I will try to cut down on the ENV variables

File: recipes/cabal/meta.yaml (hunk 2 above)
Comment (bgruening) [reply recipes/cabal/meta.yaml]: But it should be enough in the `host` section.

Comment (eggzilla) [reply recipes/cabal/meta.yaml]: Ah, ok :-) I will change it!

Comment (eggzilla) [reply recipes/cabal/meta.yaml]: Ok, removed the compiler flags

**Comment by bgruening** [comment]
Merci!

**Pull request merged**